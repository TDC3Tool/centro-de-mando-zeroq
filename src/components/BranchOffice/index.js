import React from 'react'
import { FaRegUser, FaRegClock } from 'react-icons/fa'
import PropTypes from 'prop-types'

export const BranchOffice = ({ name, persons, duration, online }) => (
  <div className='column is-12-mobile is-4-tablet is-3-widescreen'>
    <div className={`card card-sucursal card-${online}`}>
      <div className='card-content'>
        <div className='media-content'>
          <p className='title is-4 has-text-weight-bold'>{name}</p>
        </div>
      </div>
      <footer className='card-footer'>
        <div className='card-info'>
          <span className='icon is-left'>
            <FaRegUser />
          </span>
          <span className='card-info__data has-text-weight-semibold'>{persons}</span>
        </div>
        <div className='card-info'>
          <span className='icon is-left'>
            <FaRegClock />
          </span>
          <span className='card-info__data has-text-weight-semibold'>{duration}</span>
        </div>
      </footer>
    </div>
  </div>
)

BranchOffice.propTypes = {
  name: PropTypes.string.isRequired,
  persons: PropTypes.number.isRequired,
  duration: PropTypes.string.isRequired,
  online: PropTypes.bool.isRequired
}
