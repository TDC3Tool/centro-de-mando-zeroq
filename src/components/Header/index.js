import React from 'react'
import { HeaderContainer } from './styles'
import Logo from '../../assets/logo.png'

export const Header = () => (
  <HeaderContainer>
    <img src={Logo} alt='Control' />
  </HeaderContainer>
)
