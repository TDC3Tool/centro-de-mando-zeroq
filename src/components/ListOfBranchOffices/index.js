import React from 'react'
import { BoxList } from './styles'

export const ListOfBranchOffices = ({ children }) => {
  return (
    <BoxList>
      <div className='container'>
        <div className='columns is-centered is-mobile'>
          <div className='column is-10 is-12-mobile'>
            <div className='columns is-multiline is-mobile is-desktop is-tablet'>
              {children}
            </div>
          </div>
        </div>
      </div>
    </BoxList>
  )
}
