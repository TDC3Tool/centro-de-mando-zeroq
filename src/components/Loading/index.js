import React from 'react'
import { Roller } from './styles'

export const Loading = () => (
  <>
    <Roller><div /><div /><div /><div /><div /><div /><div /><div /></Roller>
  </>
)
