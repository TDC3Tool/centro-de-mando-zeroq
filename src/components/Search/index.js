import React from 'react'
import { SearchBox } from './styles'
import { FaSearch } from 'react-icons/fa'

export const Search = ({ children }) => {
  return (
    <SearchBox>
      <div className='container px-2'>
        <div className='columns is-centered is-mobile'>
          <div className='column is-10'>
            <div className='columns is-mobile is-tablet'>
              <div className='column is-12-mobile is-12-desktop is-5-widescreen'>
                <div className='field'>
                  <div className='control has-icons-left'>
                    {children}
                    <span className='icon is-small is-left'>
                      <FaSearch />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </SearchBox>
  )
}
