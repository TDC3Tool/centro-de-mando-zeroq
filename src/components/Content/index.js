import React from 'react'
import { Search } from '../Search'
import { ListOfBranchOffices } from '../ListOfBranchOffices'
import { BranchOffice } from '../BranchOffice/index'
import { useData } from '../../hooks/useData'
import { Loading } from '../Loading'

export const Content = () => {
  const {
    dataFiltered,
    isLoading,
    isError,
    query,
    _handleChange,
    isResults
  } = useData()

  return (
    <>
      <Search>
        <input
          className='input'
          placeholder='Buscar sucursal'
          type='text'
          value={query}
          onChange={_handleChange}
        />
      </Search>
      <ListOfBranchOffices>
        {isError && <div>Something went wrong ...</div>}
        {
          isLoading
            ? (
              <>
                <div className='column is-12'>
                  <div className='columns is-centered is-mobile'>
                    <div className='column is-1-tablet is-4-mobile'>
                      <Loading />
                    </div>
                  </div>
                </div>
              </>
            )
            : (
              <>
                {
                  isResults
                    ? (
                      dataFiltered.map((sucursal, inc) => (
                        <BranchOffice
                          key={inc}
                          name={sucursal.name}
                          persons={sucursal.persons}
                          duration={sucursal.duration}
                          online={sucursal.online}
                        />
                      ))
                    )
                    : (
                      <>
                        <div className='column'>
                          <div className='columns is-centered is-mobile'>
                            <div className='column is-5-tablet is-10-mobile'>
                              <h2 className='title has-text-centered has-text-white'>Sin resultados</h2>
                            </div>
                          </div>
                        </div>
                      </>
                    )

                }
              </>
            )
        }
      </ListOfBranchOffices>
    </>
  )
}
