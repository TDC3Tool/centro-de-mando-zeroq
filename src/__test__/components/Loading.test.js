import React from 'react'
import { mount } from 'enzyme'
import { create } from 'react-test-renderer'
import { Loading } from '../../components/Loading'

describe('<Loading /> component', () => {
  const loading = mount(<Loading />)

  it('Render', () => {
    expect(loading.length).toEqual(1)
  })
})

describe('<Loading /> spnapshot', () => {
  it('Check the UI of <Loading />', () => {
    const loading = create(<Loading />)
    expect(loading.toJSON()).toMatchSnapshot()
  })
})
