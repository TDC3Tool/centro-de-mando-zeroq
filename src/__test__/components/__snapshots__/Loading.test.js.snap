// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> spnapshot Check the UI of <Loading /> 1`] = `
<div
  className="sc-AykKC fPQizN"
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
